components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -728,8 +733,20 @@ void RewardsServiceImpl::RunTask(
 }
 
 void RewardsServiceImpl::TriggerOnWalletInitialized(int error_code) {
+  // webui
   for (auto& observer : observers_)
     observer.OnWalletInitialized(this, error_code);
+
+  // extension
+  EventRouter* event_router = EventRouter::Get(profile_);
```

Comment:
this code should go in brave_rewards_api.h and be triggered by `observer.OnWalletInitialized`
also you need to deal with the error code because this doesn't necessarily mean that the wallet has been created

Reply:
will be addressed in the follow up as discussed